phoenicis-scripts/src/main/java/org/phoenicis/scripts/ui/UiQuestionFactory.java

```diff
@@ -18,17 +18,37 @@
 
 package org.phoenicis.scripts.ui;
 
+import com.google.common.util.concurrent.Runnables;
+
+import java.util.List;
+
 public interface UiQuestionFactory {
     /**
-     * creates a question UI (yes/no decision)
-     * @param questionText question
-     * @param yesCallback called when user selects "yes"
-     * @param noCallback called when user selects "no"
+     * Creates a question UI (yes/no decision)
+     *
+     * @param questionText The question text
+     * @param yesCallback The callback for when the user selects "yes"
+     * @param noCallback The callback for when the user selects "no"
      */
     void create(String questionText, Runnable yesCallback, Runnable noCallback);
 
+    /**
+     * Creates a question UI (yes/no decision)
+     *
+     * @param questionText The question text
+     * @return True if the user selects "yes", false if the user selects "no"
+     */
+    boolean create(String questionText);
+
+    /**
+     * Creates a question UI (multiple-choice decision)
+     * @param questionText The question text
+     * @param choices The choices
+     * @return The selected choice or null if the user canceled the question
+     */
+    String create(String questionText, List<String> choices);
```

Comment:
Should we return the selected choice or its index?

Reply:
How is the menu handled?

Reply:
As far as I know it returns the input string. The problem with this approach are translations. If the user can select a localized string it is better to work with indices.

Reply:
I remember some scripts which use two arrays (one with and one without translation) to circumvent this issue. Really not that nice.